### PR TITLE
Lighting Revamp/Directional Part Uno

### PR DESCRIPTION
**Pull request by rgarmsen2295**
This adds support for (up to 10!) directional lights (no point lights for now, but the framework is there).

Rewrote the phong shader to look nicer (ignore the dummy shaders they will go bye bye after level remake)

There's a better way to pass all the lights in glsl but that technique can be used with other stuff too apparently so I'll wait to do a bigger refactor of shader stuff

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -30,6 +30,9 @@ static void initMisc() {
 
     // Enable z-buffer test
     glEnable(GL_DEPTH_TEST);
+
+    // Seed the PRNG with the current time for any random elements in the world
+    std::srand(std::time(NULL));
```
Comment (joncatanio) [inline src/main.cpp:35]: hmm when did this get removed? I thought there was a seed somewhere.

**Review by joncatanio — APPROVED** [review]

**Comment by joncatanio** [comment]
Gonna merge this so we can get going for tomorrow

File: src/main.cpp (hunk 1 above)
Comment (rgarmsen2295) [reply src/main.cpp:35]: I moved it into main from GameWorld is all, not really used anymore anyway, but if we load another level we don't need to reseed anything so I moved it.

File: resources/phong_vert.glsl [hunk 2]
```diff
@@ -6,16 +6,21 @@ layout(location = 1) in vec3 vertNor;
...
-	worldPos = (V * M * vertPos).xyz;
-
-	vec3 vertNormal = normalize(vertNor);
-	fragNor = normalize((tiMV * vec4(vertNormal, 0.0)).xyz);
 
+	// Set the position of the vertex in homogeneous space
 	gl_Position = P * V * M * vertPos;
+
+	// Calculate the position of the vertex in camera/view space
+	positionInCamSpace = (V * M * vertPos).xyz;
```
Comment (joncatanio) [inline resources/phong_vert.glsl:22]: Why multiply by V?

**Pull request merged**